### PR TITLE
Fixed-Task 5791: Group task - Org registration

### PR DESCRIPTION
**Pull request by saranyadevi-bc**

### CODE REVIEW

File: api/CcsSso.Core.Service/External/UserProfileService.cs [hunk 1]
```diff
@@ -355,6 +359,20 @@ public async Task<UserEditResponseInfo> CreateUserAsync(UserProfileEditRequestIn
         IsRegisteredInIdam = isRegisteredInIdam
       };
     }
+    private async Task AddUserToDefaultGroup(int ciiOrgId, int UserId)
```
Comment (brijeshpatel-bc) [inline api/CcsSso.Core.Service/External/UserProfileService.cs:362]: private async Task AddUserToDefaultGroup(int orgId, int userId)

File: api/CcsSso.Core.Service/External/UserProfileService.cs [hunk 2]
```diff
@@ -227,6 +227,10 @@ public async Task<UserEditResponseInfo> CreateUserAsync(UserProfileEditRequestIn
       _dataContext.Party.Add(party);
 
       await _dataContext.SaveChangesAsync();
+      if(isNewOrgAdmin==true)
+      {
+          await AddUserToDefaultGroup(organisation.Id, party.User.Id);
```
Comment (brijeshpatel-bc) [inline api/CcsSso.Core.Service/External/UserProfileService.cs:232]: AddUserToDefaultAdminGroup

File: api/CcsSso.Core.Service/External/UserProfileService.cs [hunk 3]
```diff
@@ -355,6 +359,20 @@ public async Task<UserEditResponseInfo> CreateUserAsync(UserProfileEditRequestIn
         IsRegisteredInIdam = isRegisteredInIdam
       };
     }
+    private async Task AddUserToDefaultGroup(int ciiOrgId, int UserId)
+    {
+        var DefaultAdminGroup = await _dataContext.OrganisationUserGroup.FirstOrDefaultAsync(x => x.OrganisationId == ciiOrgId && x.GroupType == (int)GroupType.Admin && !x.IsDeleted);
```
Comment (brijeshpatel-bc) [inline api/CcsSso.Core.Service/External/UserProfileService.cs:364]: DefaultAdminGroup --> defaultAdminGroup

File: api/CcsSso.Core.Service/External/UserProfileService.cs [hunk 4]
```diff
@@ -355,6 +359,20 @@ public async Task<UserEditResponseInfo> CreateUserAsync(UserProfileEditRequestIn
...
+        if (DefaultAdminGroup != null)
+        {
+           var UserGroupMembership = new UserGroupMembership
```
Comment (brijeshpatel-bc) [inline api/CcsSso.Core.Service/External/UserProfileService.cs:367]: UserGroupMembership  -> userGroupMembership

File: api/CcsSso.Core.Service/OrganisationService.cs [hunk 5]
```diff
@@ -354,14 +355,32 @@ private async Task CreateOrganisationAsync(OrganisationRegistrationDto organisat
...
+        }
       catch (Exception)
       {
         await _ciiService.DeleteOrgAsync(ciiOrgId);
         throw;
       }
     }
+        public async Task OrganisationAdminGroup(OrganisationRegistrationDto organisationRegistrationDto, string ciiOrgId)
+        {
+            var Organisation = await _dataContext.Organisation.Include(o => o.UserGroups).FirstOrDefaultAsync(o => !o.IsDeleted && o.CiiOrganisationId == ciiOrgId);
```
Comment (brijeshpatel-bc) [inline api/CcsSso.Core.Service/OrganisationService.cs:368]: Organisation --> organisation

File: api/CcsSso.Core.Service/OrganisationService.cs [hunk 6]
```diff
@@ -354,14 +355,32 @@ private async Task CreateOrganisationAsync(OrganisationRegistrationDto organisat
...
+            var IsDefaultAdminGroup = await _dataContext.OrganisationUserGroup.AnyAsync(x => x.OrganisationId == Organisation.Id && x.GroupType == (int)GroupType.Admin && !x.IsDeleted);
```
Comment (brijeshpatel-bc) [inline api/CcsSso.Core.Service/OrganisationService.cs:369]: IsDefaultAdminGroup - isDefaultAdminGroupExists

File: api/CcsSso.Core.Service/OrganisationService.cs [hunk 7]
```diff
@@ -354,14 +355,32 @@ private async Task CreateOrganisationAsync(OrganisationRegistrationDto organisat
...
 
+            if (!IsDefaultAdminGroup)
+            {
+                var group = new OrganisationUserGroup
+                {
+                    OrganisationId = Organisation.Id,
+                    UserGroupName = "Organisation Administrators",
+                    UserGroupNameKey = "ORGANISATION_ADMINISTRATORS",
```
Comment (brijeshpatel-bc) [inline api/CcsSso.Core.Service/OrganisationService.cs:377]: move group name and key to constant file.

**Review by brijeshpatel-bc — CHANGES_REQUESTED** [review]

File: api/CcsSso.Core.Service/External/UserProfileService.cs (hunk 1 above)
Comment (saranyadevi-bc) [reply api/CcsSso.Core.Service/External/UserProfileService.cs]: Updated the method name and parameter to AddUserToDefaultGroup(int orgId, int userId)

File: api/CcsSso.Core.Service/External/UserProfileService.cs (hunk 2 above)
Comment (saranyadevi-bc) [reply api/CcsSso.Core.Service/External/UserProfileService.cs]: Updated the method name to AddUserToDefaultAdminGroup

File: api/CcsSso.Core.Service/External/UserProfileService.cs (hunk 3 above)
Comment (saranyadevi-bc) [reply api/CcsSso.Core.Service/External/UserProfileService.cs]: Updated to defaultAdminGroup

File: api/CcsSso.Core.Service/External/UserProfileService.cs (hunk 4 above)
Comment (saranyadevi-bc) [reply api/CcsSso.Core.Service/External/UserProfileService.cs]: Updated

File: api/CcsSso.Core.Service/OrganisationService.cs (hunk 5 above)
Comment (saranyadevi-bc) [reply api/CcsSso.Core.Service/OrganisationService.cs]: updated

File: api/CcsSso.Core.Service/OrganisationService.cs (hunk 6 above)
Comment (saranyadevi-bc) [reply api/CcsSso.Core.Service/OrganisationService.cs]: Updated

File: api/CcsSso.Core.Service/OrganisationService.cs (hunk 7 above)
Comment (saranyadevi-bc) [reply api/CcsSso.Core.Service/OrganisationService.cs]: Moved group name and key to constant file.

**Review by brijeshpatel-bc — APPROVED** [review]

**Pull request merged**